[test] Add component tests for some Vue Widget components

christian-byrne opened the pull request:
## Summary

Adds component tests for some Vue Widget components.

## Changes

- **What**: Added tests
- Changes include path for component tests

## Review Focus

- whether the tests are getting good coverage and applying best-practices

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5409-test-Add-component-tests-for-some-Vue-Widget-components-2666d73d3650819b9b0cf484031a0bdb) by [Unito](https://www.unito.io)

christian-byrne committed:
add component tests for vue widgets

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetInputText.test.ts:
The tests in here look pretty good, but it's kind of hard to see what differentiates one from another when there's so much similarity in the arrange/act sections

You already have a convenience setup function for the widget data, what about one for the wrapper too?

DrJKL replied:
You could probably have a similar utility to batch the operations like finding the child element, setting value, then blurring. If that's the standard, anything that deviates will stand out as interesting.

christian-byrne replied:
Moved to utilities in [77b0207](https://github.com/Comfy-Org/ComfyUI_frontend/pull/5409/commits/77b0207cb3a24937562f8c4832ccf835f63911de)

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetInputText.test.ts:
```diff
@@ -0,0 +1,297 @@
+      })
+
+      const input = wrapper.find('input[type="text"]')
+      expect((input.element as HTMLInputElement).disabled).toBe(true)
```
This could be inside an `instanceof` gate

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetSlider.test.ts:
```diff
@@ -0,0 +1,150 @@
+    callback
+  })
+
+  const mountComponent = (
```
Oh! Like this!

DrJKL commented on src/renderer/extensions/vueNodes/widgets/components/WidgetSlider.test.ts:
```diff
@@ -0,0 +1,150 @@
+      const widget = createMockWidget(42)
+      const wrapper = mountComponent(widget, 42)
+
+      const input = wrapper.find('input[type="number"]')
+      expect((input.element as HTMLInputElement).value).toBe('42')
```
If you moved these into an assertion utility, you could explicitly check that `input.element instanceof HTMLInputElement` and throw otherwise.

christian-byrne replied:
Done in [77b0207](https://github.com/Comfy-Org/ComfyUI_frontend/pull/5409/commits/77b0207cb3a24937562f8c4832ccf835f63911de)

christian-byrne committed:
[refactor] improve widget test readability and type safety - addresses @DrJKL's review feedback

- Add mountComponent utility function for consistent test setup
- Add setInputValueAndTrigger helper to batch common test operations
- Replace type assertions with proper instanceof checks for type safety
- Remove duplicate test setup code to improve test readability
- Fix TypeScript errors in WidgetSlider tests

These changes address all review comments by making tests easier to read
and understand while ensuring proper type checking.

christian-byrne commented on src/renderer/extensions/vueNodes/widgets/components/WidgetSlider.test.ts:
```diff
@@ -83,37 +93,8 @@ describe('WidgetSlider Value Binding', () => {
-      const wrapper = mountComponent(widget, 42)
-
-      const input = wrapper.find('input[type="number"]')
-      expect((input.element as HTMLInputElement).value).toBe('42')
```
Removed duplicate tests

christian-byrne committed:
[refactor] apply consistent test patterns to WidgetSelect.test.ts

- Add mountComponent utility function for consistent test setup
- Add setSelectValueAndEmit helper to batch select operations
- Remove repetitive mount boilerplate code throughout tests
- Maintain existing test coverage while improving readability

This ensures all widget component tests follow the same patterns
established in WidgetInputText and WidgetSlider tests.
